site/_js/web-components/tag-pill-list.js

```diff
@@ -23,44 +23,77 @@ import {html} from 'lit-element';
 import {unsafeSVG} from 'lit-html/directives/unsafe-svg';
 import closeIcon from '../../_includes/icons/close.svg';
 
+import {store} from '../store';
+import {removeEntry} from '../actions/filter';
+
 export class TagPillList extends BaseElement {
+  static get properties() {
+    return {
+      items: {type: Array, reflect: true},
+    };
+  }
+
   constructor() {
     super();
-
     this.items = [];
   }
 
-  static get properties() {
-    return {
-      items: {type: Array, reflect: true},
-    };
+  connectedCallback() {
+    super.connectedCallback();
+    store.subscribe(this.onStoreUpdate.bind(this));
   }
 
-  _handleClick(item) {
-    this.items = this.items.filter(i => i !== item);
+  onStoreUpdate(state) {
+    const filters = state.filters || {};
+    const items = [];
+    for (const [name, entries] of Object.entries(filters)) {
+      for (const item of entries) {
+        items.push({
+          name: name,
+          value: item.value,
+          label: item.label,
+        });
+      }
+    }
+    this.items = items;
+  }
 
-    const event = new CustomEvent('removed-pill', {
-      detail: item,
-    });
+  _onClickPill(item) {
+    removeEntry(item.name, item);
+  }
 
-    this.dispatchEvent(event);
+  _onClickClearPills() {
+    for (const item of this.items) {
+      removeEntry(item.name, item);
```

Comment:
I think it's worth implementing a `clearFilters` action instead, that simply sets store.filters back to an empty object - would also be less costly.